src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -245,11 +270,37 @@ private SafeCurlHandle CreateRequestHandle(RequestCompletionSource state)
             }
 
             SetProxyOptions(state, requestHandle);
+            SetRequestHandleDecompressionOptions(requestHandle);
             // TODO: Handle headers and other options
 
             return requestHandle;
         }
 
+        private void SetRequestHandleDecompressionOptions(SafeCurlHandle requestHandle)
+        {
+            var curlEncoding = new StringBuilder();
+            if (AutomaticDecompression != DecompressionMethods.None)
+            {
+                var needComma = false;
+                if ((AutomaticDecompression & DecompressionMethods.GZip) != 0)
+                {
+                    curlEncoding.Append(EncodingNameGzip);
+                    needComma = true;
+                }
+
+                if ((AutomaticDecompression & DecompressionMethods.Deflate) != 0)
+                {
+                    if (needComma)
+                    {
+                        curlEncoding.Append(',');
+                    }
+
+                    curlEncoding.Append(EncodingNameDeflate);
+                }
+                Interop.libcurl.curl_easy_setopt(requestHandle, CURLoption.CURLOPT_ACCEPTENCODING, curlEncoding.ToString());
+            }
+        }
+
```

Comment:
Sending out a PR to remove the string builder allocation. Thank you, @stephentoub  and @justinvp 

@stephentoub :
> The Windows implementation looks like there might be some validation done in winhttp that causes an exception to be thrown

Winhttp throws error when the native library does not support the feature. Curl-Option validation is getting checked-in via other PR #2597.  Depending on which of these go first, one of us will incorporate the error checking here.